### ISSUE TITLE
Update definition of "error"

### ISSUE THREAD

**Issue by samm82** [samm82/TestGen-Thesis#34]
My current definition of "error" as "a human action that produces an incorrect result" is a little incomplete and misleading (at least to me). Tracking the original source led me to IEEE2010, which has the more complete definition of "a human action that produces an incorrect result, such as software containing a fault", although it also defines it as "an incorrect step, process, or data definition" and "an incorrect result".

- [x] Update definition of "error" in notes
- [x] Investigate definitions of "fault" and "failure"
- [x] Add IEEE2010 to the GitLab repo
- [x] Move "defect" notes from research to glossary
- [x] Revisit related test approaches: 
   - [x] error guessing
   - [x] ~~error~~ fault seeding
   - [x] error tolerance testing
   - [x] fault-based testing, etc.
- [x] Investigate "malfunction"
- ~Use these discrepancies as "core examples"? (#131)~ Moved to #132

**Comment by JacquesCarette** [comment]
I find these definitions of "error" really odd. In a sense, it is a weird halfway house between dictionary definitions (https://www.merriam-webster.com/dictionary/error) and software-specific definitions.

Basically, I fail to see this definition as serving a useful purpose. If they want generality, they ought to re-use a dictionary definition. If they want specificity, then they should actually do that!

**Comment by samm82** [comment]
Agreed! Also, since it defines "error" as both "a human action that produces an incorrect result" and "an incorrect result", it implicitly defines "error" as "a human action that produces an error"! 👀

This is probably the opening to yet another rabbit hole that I'll try not to fall into; this observation may just manifest as a footnote or comment about the state of precise terminology (i.e., the lack thereof) in software engineering. I found this because a friend in the Software Testing course asked me about the difference between an error, fault, and failure and I realized I had some definitions tucked away in my notes and went on a side quest to learn more....

**Comment by samm82** [comment]
All items addressed; this issue can be closed!
